### PR TITLE
*BSD fixes

### LINKED ISSUE #3721: Test failing on OpenBSD 7.1
The test named `Check offline sniff() with a PcapNg file and a filter (by file object)`  is failing on OpenBSD 7.1 with the following error `scapy.error.Scapy_Exception: Failed to compile filter expression udp (-1)`. I will try to reproduce the issue on a local VM.

__Originally posted by @guedou in https://github.com/secdev/scapy/pull/3716#issuecomment-1220521197__

### PR DESCRIPTION
**Pull request by guedou**
This PR adds fixes for FreeBSD, OpenBSD and NetBSD.

It fixes #3721.

### CODE REVIEW

File: .config/ci/test.sh [hunk 1]
```diff
@@ -30,11 +30,21 @@ then
...
+elif [[ "$OSTYPE" = "darwin"* ]] || [ "$TRAVIS_OS_NAME" = "osx" ] || [[ "$OSTYPE" = "FreeBSD" ]] || [[ "$OSTYPE" = *"bsd"* ]]
 then
   OSTOX="bsd"
   # Travis CI in macOS 10.13+ can't load kexts. Need this for tuntaposx.
   UT_FLAGS+=" -K tun -K tap"
+  if [[ "$OSTYPE" = "openbsd"* ]]
+  then
+    # Note: LibreSSL 3.6.* does not support X25519 according to
+    # the cryptogaphy module source code
+    UT_FLAGS+=" -K libressl"
```
Comment (gpotter2) [inline .config/ci/test.sh:42]: I think that in UTscapy, tags are expected to indicate what the test works on rather that what it doesn't work on :/

File: .config/ci/openssl.py [hunk 2]
```diff
@@ -38,8 +38,11 @@
 """
 
 # Copy and edit
-with open(OPENSSL_CONFIG, 'rb') as fd:
-    DATA = fd.read()
+try:
+    with open(OPENSSL_CONFIG, 'rb') as fd:
+        DATA = fd.read()
+except FileNotFoundError:
+    DATA = b""
```
Comment (gpotter2) [inline .config/ci/openssl.py:45]: It doesn't make sense to continue if this fails.
```suggestion
    print("WARNING: No OpenSSL config file found.", file=sys.stderr)
    sys.exit(1)
```

+ `import sys`

**Review by gpotter2 — COMMENTED** [review]
Looking good ! Awesome work !!

File: .config/ci/openssl.py (hunk 2 above)
Comment (guedou) [reply .config/ci/openssl.py:45]: Why is that? As we are providing our own openssl config file, everything work as expected.

Comment (gpotter2) [reply .config/ci/openssl.py:45]: Oh you're right. I was thinking it made no sense to provide it if for instance libressl was used, but you're right there could be a case with no openssl config (or we couldn't find it).

File: .config/ci/test.sh (hunk 1 above)
Comment (guedou) [reply .config/ci/test.sh:42]: To be, these flags are used to tag `features`, and later on _kill_ them with `-K ` or a config file.

Comment (guedou) [reply .config/ci/test.sh:42]: Your call.

**Pull request merged**